modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/DLAdminDisplayContext.java

```diff
@@ -251,7 +251,9 @@ public String getRememberCheckBoxStateURLRegex() {
 	}
 
 	public long getRepositoryId() {
-		if (_repositoryId != null) {
+		_repositoryId = _dlPortletInstanceSettings.getRepositoryId();
+
+		if (_repositoryId != 0) {
 			return _repositoryId;
 		}
```

Comment:
Why can't we cache the value and avoid calling settings each time? I think something like this would be better:
```
if (_repositoryId != null) {
    return _repositoryId;
}

long repositoryId = _dlPortletInstanceSettings.getRepositoryId();

if (_repositoryId != 0) {
    _repositoryId = repositoryId;

    return _repositoryId;
}

// The rest is the same
```